# Conception d'un moulin à eau
## Correction

<img src="moulin.png" width="500" style="float:center">

**Cahier des charges et contraintes technologiques**



Caractéristiques 
* Vitesse nominale de rotation de la roue à aubes : $N_{1}=30$ tr/mn
* Vitesse nominale de rotation de l'arbre intermédiare : $N_{2}=180$ tr/mn
* Couple nominale mesuré sur l'axe de la roue à aube : $C_1=3200$ Nm 
* Masse de l'arbre d'entrée: $M=1000$ Kg 
* Rendement du réducteur : $r=0.98$
* Encombrement de l'arbre d'entrée: $d_1mini=50$ mm et $d_1maxi=110$ mm
* Encombrement de l'arbre intermédiaire: $d_2mini=15$ mm et $d_2maxi=50$ mm
* $h=200$ mm, $c=3000$ mm et $e=3600$ mm 
* Centre de gravité de l'arbre d'entrée en G,  $\vec{AG}=0.5e\vec{x}$ 
* Diamètre de la roue à aube mesuré au centre d'une aube : $D_e=4$ m 

Données sur les engrenages

<img src="donnee_engrenageM.png" width="700" style="float:center">

### 1.1 Train calcul du train d'engrenage

**Question 1):**

Calculez l'angle de pression de fonctionnement, $\alpha_1$, et le module de fonctionnement, $m_1$ du train d'engrenage

<img src="entraxe_deport1.png" width="300" style="float:left">
<img src="entraxe_deport2.png" width="300" style="float:left">

L'entraxe de fonctionnement $a$ s'écrit avec l'angle de pression en fonctionnement $\alpha$ :
$$a = \frac{r_{b1}}{\cos\alpha} + \frac{r_{b2}}{\cos\alpha}$$
L'entraxe de taillage $a_0$ s'écrit avec l'angle de pression de taillage $\alpha_0$ :
$$a_0 = \frac{r_{b1}}{\cos\alpha_0} + \frac{r_{b2}}{\cos\alpha_0}$$

$\DeclareMathOperator{\inv}{inv}$
<div class="alert alert-info">     
Engrenement sans jeux :
    $$\inv(\alpha)-\inv(\alpha_0)=2\tan\alpha_0\dfrac{x_1+x_2}{Z_1+Z_2}$$

Avec:
    $$\inv(x)=\tan(x)-x\text{ et $x$ en radians}$$    
Relation entre le module de taillage $m_0$ et de fonctionnement $m_1$ :
    $$m\cos\alpha=m_0\cos\alpha_0$$   
</div>

<div class="alert alert-info" >   
Formule des rayons primitifs $r_1$ et $r_2$ respectif des roues E1 et E2
    $$r_i=\dfrac{m}{2}Z_i$$    
    
Formule de l'entraxe  :
    $$a=r_1+r_2=\dfrac{m}{2}\left(Z_{1}+Z_{2}\right)$$
    donc $$Z_{1}+Z_{2}=\dfrac{2a}{m}$$
    
La somme des dentures peut alors s'exprimer sous la forme :    
    $$Z_{1}+Z_{2}=\dfrac{2a\cos\alpha}{m_{0}\cos\alpha_0}$$ 
    
Angle de pression de fonctionnement  $\alpha$ s'obtient par la résolution de la fonction non linéaire suivante:
    $$\inv(\alpha) - \inv(\alpha_0) - 2\tan\alpha_0\dfrac{(x_1+x_2)m_{0}\cos\alpha_0}{2a\cos\alpha}=0$$  
    
Dans le cas de cet exercice la somme des déports est nulle donc la résolution devient plus simple:
   $$x_1+x_2=0$$  
   $$\inv(\alpha) = \inv(\alpha_0)$$  

Donc
  $$\alpha = \alpha_0$$  et $$m=m_0$$
    
</div>

In [16]:
import numpy as np
alpha0=20
alpha = alpha0
m0=16
m=m0

print('alpha : %.1f °'%alpha)
print('m : %.1f mm'%m)

alpha : 20.0 °
m : 16.0 mm


**Question 2):**

Calculez les rayons primitifs de fonctionnement et le nombre de dents des deux roues d'engrenage $E_1$ et $E_2$.

<div class="alert alert-info" >   
Rappel sur le rapport de réduction :
    
Roulement sans glissement au point de contact $I$ sur les rayons primitifs : $\omega_1 r_1 = \omega_2 r_2$.
    
Donc :    
    $$i=\dfrac{\omega_2}{\omega_1}=\dfrac{N_2}{N_1}=\dfrac{Z_1}{Z_2}=\dfrac{r_1}{r_2}$$        
</div>

Les rayons primitifs s'obtient à partir de l'entraxe $a$ et du rapport de réduction $i$:


$$r_1 + r_2 = a=a_0 $$


$$r_1=\dfrac{ai}{1+i}\quad r_2=\dfrac{a}{1+i}$$

In [17]:
a  = 1120
N1 = 30
N2 = 180
i = N2/N1
r1 =a*i/(1+i)
r2 =a/(1+i)
print('rapport de réduction i=%.2f'%i)
print('r1= %.2f mm et r2= %.2f mm'%(r1,r2))

rapport de réduction i=6.00
r1= 960.00 mm et r2= 160.00 mm


Le nombre de dents de la roue $E_i$ (attention il est parfois necessaire de respecter la partie entière)
$$ Z_i=\frac{2r_i}{m}$$

In [18]:
Z1=2*r1/m
Z2=2*r2/m

print('Z1= %d   Z2=%d'%(Z1,Z2))

Z1= 120   Z2=20


**Question 3):**

Estimez la valeur de la contrainte au pied de dent pour $E_1$:

<div class="alert alert-info" > 
Contrainte en pieds de dent (formule approchée) :
    $$\sigma_{max}=\frac{5.5T_{max}}{b m}$$
</div>    

L'effort tangentiel maximum $T_{max}$ s'obtient à partir du couple (moteur) sur la roue à aubes $C_1$
$$ T_{max}=\frac{C_1}{r_1}$$

In [19]:
b  = 200.   # Largeur en mm
C1 = 3200.  # Couple en N.m
Tmax = C1*1e3/r1  # Effort tangentiel en N
sigFlmax = 5.5*Tmax/(b*m)
print('Tmax= %.2f N'%Tmax)
print('sigFlmax= %.2f MPa'%sigFlmax)

Tmax= 3333.33 N
sigFlmax= 5.73 MPa


**Question 4):**

estimez la valeur de la contrainte au pied de dent pour $E_1$ à l'aide de la méthode approchée

<div class="alert alert-info" > 
Pression maxi de Hertz : $$\mathcal{P}_{Hmax}=\sqrt{\frac{T_{max}}{b\cos\alpha}\frac{r_1+r_2}{r_1r_2\sin\alpha}\frac{E^*}{\pi}}\quad E^*=\frac{E}{2(1-\nu^2)}$$
</div>

In [20]:
E=210000
mu=0.3
Eet=E/(2*(1-mu*mu))
PHmax = np.sqrt(Tmax*(r1+r2)*Eet/(b*np.cos(alpha*np.pi/180)*r1*r2*np.sin(alpha*np.pi/180)*np.pi))
print('PHmax=%.2f MPa'%PHmax)

PHmax=117.85 MPa


**Question 5):**

Choisissez un matériau compatible avec les valeurs trouvées au deux questions précédentes.

<div class="alert alert-success" > 
Avec des dentures en AF42 $\sigma_{Flim}=125$MPa > 5.73 MPa  et $\sigma_{Hlim}=315$MPa > 118 MPa -> OK !!
</div>

**Question 6):**

Estimez numériquement l'équilibrage à l'usure des roues $E_1$ et $E_2$. Proposez une solution
si l'équilibrage n'est pas satisfaisant.

<div class="alert alert-info" > 
Rappel de cours :
</div>

<img src="conduite_engrenement.png" width="600" style="float:center">

<div class="alert alert-info"> 
Le rapport de conduite est la grandeur sans dimension :
$$C=\varepsilon_\alpha=\dfrac{AI}{p_b} + \dfrac{IB}{p_b}$$
<br>
Avec :
    <li>$AI=\sqrt{(r_1+h_{a1})^2-(r_1\cos\alpha)^2}-r_1\sin\alpha$ : Longueur d'appproche dans le mvt d'engrènement.</li><br>
    <li>$IB=\sqrt{(r_2+h_{a2})^2-(r_2\cos\alpha)^2}-r_2\sin\alpha$  : Longueur de retrait dans le mvt. d'engrènement.</li><br>
    <li>$p_b = \pi m\cos\alpha$ : Pas de base, i.e distance entre 2 flancs consécutifs de la denture prise sur le diamètre de base.</li><br>
<br>
Le rapport de conduite indique le nombre de dents simultanément en contact lors de l'engrènemenet, il ne doit jamais être inférieur à 1. En pratique on essaie d'avoir $\varepsilon_\alpha \geq \simeq 1.3$.
</div>

<div class="alert alert-info">
Au cours de l'engrènement il y a entre la dent menante et la menée :
<li>Roulement et glissement sur les longueurs d'approche et de retrait, ce glissement est maximal en $A$ et en $B$.</li>
<li>Roulement sans glissement au point de contact $I$.</li>
<br>
L'usure est fonction de la vitesse de glissement et de la pression de contact, pour les engrenanges elle est inévitable, en pratique on cherche à la limiter et à équilibrer les usures de la roue et pignon, c'est à dire à égaliser les glissements spécifiques maximuns de la roue et du pignon.
<br>
<br>
On compare :
 <li>Pour roue : $\dfrac{AI}{r_2\sin\alpha}$.</li>
 <li>Pour le pignon : $\dfrac{IB}{r_1\sin\alpha}$.</li>
</div>

In [41]:
alpha = 20
calpha=np.cos(alpha*np.pi/180)
salpha=np.sin(alpha*np.pi/180)

# Rayons sur le pignon :
x1=-0.25
r1 =Z1*m/2
rb1 = r1*calpha
ha1 = m*(1+x1)
# Rayons sur la roue :
x2 = 0.25
r2 =Z2*m/2
rb2 = r2*calpha
ha2 = m*(1+x2)
# Pas de base :
pb = np.pi*m*calpha
# Longueur d'approche et de retrait :
IH1 = r1*salpha
IH2 = r2*salpha
AI = np.sqrt((r1+ha1)**2 -rb1**2)-IH1
IB = np.sqrt((r2+ha2)**2 -rb2**2)-IH2
# Calcul des glissements :
Gs1 = AI/IH2
Gs2 = IB/IH1
print('Gliss. spec. maxi roue(1) :%.4g  roue(2) :%.4g'%(Gs1,Gs2))
# Coefficient d'usure :
us1 = AI/pb*Z1
us2 = IB/pb*Z2
print('Coef usure   maxi roue(1) :%.4g   roue(2) :%.4g'%(us1,us2))

Gliss. spec. maxi roue(1) :0.6138  roue(2) :0.1348
Coef usure   maxi roue(1) :85.33   roue(2) :18.73


<div class="alert alert-danger" >
   Les roues d'engrenage ne sont pas équilibré à l'usure !!
</div>


<div class="alert alert-success" > 
Une solution serait de jouer sur les déports de denture mais en conservant
       $$x_1+x_2=0$$
</div>

In [54]:
# Rayons sur le pignon :
x1=-0.75
r1 =Z1*m/2
rb1 = r1*calpha
ha1 = m*(1+x1)
# Rayons sur la roue :
x2 = -x1
r2 =Z2*m/2
rb2 = r2*calpha
ha2 = m*(1+x2)
# Pas de base :
pb = np.pi*m*calpha
# Longueur d'approche et de retrait :
IH1 = r1*salpha
IH2 = r2*salpha
AI = np.sqrt((r1+ha1)**2 -rb1**2)-IH1
IB = np.sqrt((r2+ha2)**2 -rb2**2)-IH2
# Rapport de conduite :
C = (AI+IB)/pb
print('Rapport de conduite : %.3f'%C)
# Calcul des glissements :
Gs1 = AI/IH2
Gs2 = IB/IH1
print('Gliss. spec. maxi roue(1) :%.4g  roue(2) :%.4g'%(Gs1,Gs2))
# Coefficient d'usure :
us1 = AI/pb*Z1
us2 = IB/pb*Z2
print('Coef usure   maxi roue(1) :%.4g   roue(2) :%.4g'%(us1,us2))     

Rapport de conduite : 1.475
Gliss. spec. maxi roue(1) :0.2105  roue(2) :0.1771
Coef usure   maxi roue(1) :29.26   roue(2) :24.62
